docs/sources/operations/query-fairness/_index.md

```diff
@@ -0,0 +1,121 @@
+or a subset of those queriers are allowed to dequeue from a tenant queue.
+
+Tenant queues are the first level of the queue hierarchy. When a tenant
+executes a query without any further controls, all of its sub-queries are
+enqueued to the first level queue.
+
+However, tenant queues can also have sub-queues. When a task (sub-request) is
+dequeued, the dequeuing algorithm of the queue picks a task round robin across
+the first level (local queue) and all its sub-queues.
+
+![Hierarchical queues](./hierarchical-queues.png)
+
+The figure above shows that a tenant queue has a local queue, which is a leaf
+node in the queue tree, and a set of sub-queues. Each sub-queue, again like the
+tenant queue, consists of a local queue, and possible sub-queues, resulting in
+a recursive tree structure.
+
+So, how can we make use of these tree-like queue structure to achieve query fairness?
+
+## How to control query fairness
+
+As already mentioned, by default, sub-queries are only enqueued at the first
+(tenant) level of the queue tree. The tenant is provided by the `X-Scope-OrgID`
+header that is required when running Loki in multi-tenant mode.
+
+The HTTP header `X-Loki-Actor-Path` can be used to control to which sub-queue a
+query (or more correctly its sub-queries) is enqueued.
+
+The following example shows a `curl` command that invokes the HTTP endpoint for range queries
+and passes both the `X-Scope-OrgID` and the `X-Loki-Actor-Path` headers.
+
+```bash
+curl -s http://localhost:3100/loki/api/v1/query_range?xxx \
+    -H 'X-Scope-OrgID: grafana' \
+    -H 'X-Loki-Actor-Path: joe'
+```
+
+The query that this request invokes ends up in the sub-queue `joe` of the
+tenant queue `grafana`. Another user can use their own name in the actor path
+header to enqueue their queries in their own sub-queue.
+
+Since the tenant queue chooses the next task in a round-robin manner, both
+actors (in our case human users) get their 50% share when a querier dequeues
+from that tenant.
+
+Even when there are tasks in the local queue of the tenant, the local queue get
+1/3 and each sub-queue gets 1/3 of their share.
+
+With N actors, each actor get 1/Nth of their share.
+
+Since explained implementation and the header name already suggest, it is
+possible to enqueue queries several levels deep. To do so, you can construct a
+path to the sub-queue using the `|` delimiter in the header value, as shown in
+the following examples.
+
+```bash
+curl -s http://localhost:3100/loki/api/v1/query_range?xxx \
+    -H 'X-Scope-OrgID: grafana' \
+    -H 'X-Loki-Actor-Path: users|joe'
+
+curl -s http://localhost:3100/loki/api/v1/query_range?xxx \
+    -H 'X-Scope-OrgID: grafana' \
+    -H 'X-Loki-Actor-Path: apps|logcli'
+```
+
+There is a limit to how deep a path and thus the queue tree can be. This is
+controlled by Loki's `-query-scheduler.max-queue-hierarchy-levels` CLI argument
+or its respective YAML configuration block:
+
+```yaml
+query_scheduler:
+  max_queue_hierarchy_levels: 3
```

Comment:
What is the default value for `max_queue_hierarchy_levels`?  If the default is `3` then let's document how this works with three levels.

Reply:
and if `3` is not recommended, then we should probably change the default value?